From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TreeKR <johanbag290@gmail.com>
Date: Mon, 27 Sep 2021 13:49:14 +0900
Subject: [PATCH] Add EntityDespawnEvent Impl


diff --git a/src/main/java/xyz/r2turntrue/poper/event/EntityDespawnEvent.java b/src/main/java/xyz/r2turntrue/poper/event/EntityDespawnEvent.java
index 48ae56dab511fe39d0359daeb188098598e5ae82..e3f8365d08ba9738bf5e03adb7f48eaa2f7c15f3 100644
--- a/src/main/java/xyz/r2turntrue/poper/event/EntityDespawnEvent.java
+++ b/src/main/java/xyz/r2turntrue/poper/event/EntityDespawnEvent.java
@@ -1,6 +1,7 @@
 package xyz.r2turntrue.poper.event;
 
 import org.bukkit.entity.Entity;
+import org.bukkit.event.Cancellable;
 import org.bukkit.event.HandlerList;
 import org.bukkit.event.entity.EntityEvent;
 import org.jetbrains.annotations.NotNull;
@@ -10,12 +11,18 @@ import org.jetbrains.annotations.NotNull;
  * the world because it has existed for 5 minutes or player is go away from the despawn range.
  * <p>
  */
-public class EntityDespawnEvent extends EntityEvent {
+public class EntityDespawnEvent extends EntityEvent implements Cancellable {
 
     private static final HandlerList handlers = new HandlerList();
+    private boolean isNatural = false;
 
-    public EntityDespawnEvent(@NotNull Entity what) {
+    public EntityDespawnEvent(@NotNull Entity what, @NotNull boolean isNatural) {
         super(what);
+        this.isNatural = isNatural;
+    }
+
+    public boolean isNatural() {
+        return isNatural;
     }
 
     @Override
@@ -26,4 +33,16 @@ public class EntityDespawnEvent extends EntityEvent {
     public static HandlerList getHandlerList() {
         return handlers;
     }
+
+    private boolean cancelled = false;
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    @Override
+    public void setCancelled(boolean cancel) {
+        cancelled = cancel;
+    }
 }
\ No newline at end of file
