From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TreeKR <johanbag290@gmail.com>
Date: Mon, 27 Sep 2021 13:07:27 +0900
Subject: [PATCH] Update Gitignore


diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000000000000000000000000000000000000..b23bde3b5e881f146539a307d0a59f2177b6a036
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,38 @@
+.gradle/
+
+# Eclipse stuff
+/.classpath
+/.project
+/.settings
+
+# netbeans
+/nbproject
+/nb-configuration.xml
+
+# we use maven!
+/build.xml
+
+# maven
+/target
+
+# vim
+.*.sw[a-p]
+
+# various other potential build files
+/build
+/bin
+/dist
+/manifest.mf
+
+# Mac filesystem dust
+.DS_Store
+
+# intellij
+*.iml
+*.ipr
+*.iws
+.idea/
+
+# vs code
+/.vscode
+/.factorypath
